lib/puppet/defaults.rb

```diff
@@ -1758,8 +1758,16 @@ def self.initialize_default_settings!(settings)
         for the node stored in puppetdb are current. However, this will double the fact
         submission load on puppetdb, so it is disabled by default.",
     },
+    :publicdir => {
+      :default  => nil,
+      :type     => :directory,
+      :mode     => "0755",
+      :desc     => "The directory where Puppet public files are stored. Generally,
+        this directory can be removed without causing harm (although it
+        might result in spurious service restarts)."
```

Comment:
I'd remove the "Generally..." comment. So long as `publicdir` is defined relative to `vardir`, then puppet will create it when applying the settings catalog (when `Puppet.settings.use` is called).